lib/Dialect/ESI/ESIPasses.cpp

```diff
@@ -180,6 +193,48 @@ circt::rtl::RTLExternModuleOp ESIRTLBuilder::declareStage() {
   return declaredStage;
 }
 
+/// Return the InterfaceType which corresponds to an ESI port type. If it
+/// doesn't exist in the cache, build the InterfaceOp and the corresponding
+/// type.
+InterfaceOp ESIRTLBuilder::getOrConstructInterface(ChannelPort t) {
+  auto ifaceIter = portTypeLookup.find(t);
+  if (ifaceIter != portTypeLookup.end())
+    return ifaceIter->second;
+  auto iface = constructInterface(t);
+  portTypeLookup[t] = iface;
+  return iface;
+}
+
+InterfaceOp ESIRTLBuilder::constructInterface(ChannelPort chan) {
+  auto ctxt = getContext();
+
+  InterfaceOp iface = create<InterfaceOp>(loc, "IDataVR");
```

Comment:
I think we won't want to hard-code the interface name in case we have many different channel types. Should it use some info from the channel type to construct the name?

Reply:
Yep. I did that in another branch and forgot that I didn't pull that code in.